firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/CheckForUpdateClient.java

```diff
@@ -135,11 +147,27 @@ private boolean isNewerBuildVersion(AppDistributionReleaseInternal latestRelease
         > getInstalledAppVersionCode(firebaseApp.getApplicationContext());
   }
 
-  private boolean isInstalledRelease(AppDistributionReleaseInternal latestRelease) {
+  @VisibleForTesting
+  boolean isInstalledRelease(AppDistributionReleaseInternal latestRelease) {
     if (latestRelease.getBinaryType().equals(BinaryType.APK)) {
-      // TODO(rachelprince): APK codehash verification. For now assume
-      // the release is identical unless the build version is different
-      return true;
+      Context context = firebaseApp.getApplicationContext();
+      PackageInfo metadataPackageInfo;
+      try {
+        metadataPackageInfo =
+            context
+                .getPackageManager()
+                .getPackageInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+      } catch (PackageManager.NameNotFoundException e) {
+        return false;
+      }
+
+      String externalCodeHash = extractApkCodeHash(metadataPackageInfo);
```

Comment:
can all of these blocks of code go in the try block? 

That way we could declare metadataPackageInfo inline instead of above the try block